x-pack/solutions/observability/plugins/synthetics/server/synthetics_service/private_location/synthetics_private_location.ts

```diff
@@ -362,7 +366,9 @@ export class SyntheticsPrivateLocation {
     allPrivateLocations: SyntheticsPrivateLocations,
     spaceId: string
   ) {
-    const soClient = this.server.coreStart.savedObjects.createInternalRepository();
+    const soClient = this.server.coreStart.savedObjects
+      .getUnsafeInternalClient()
+      .asScopedToNamespace(spaceId);
```

Comment:
Since before this PR package policy was always created in default space, this will not fetch existing package policies for a non default space. it will end up recreating those.